Client/Frontend/Browser/BrowserViewController.swift

```diff
@@ -195,6 +213,56 @@ extension BrowserViewController: WKNavigationDelegate {
+        // If the page uses window.open() or target="_blank", open the page in a new tab.
+        // TODO: This doesn't work for window.open() without user action (bug 1124942).
+        let tab = tabManager.addTab(request: navigationAction.request, configuration: configuration)
+        return tab.webView
+    }
+
+    func webView(webView: WKWebView, runJavaScriptAlertPanelWithMessage message: String, initiatedByFrame frame: WKFrameInfo, completionHandler: () -> Void) {
+        // Show JavaScript alerts.
+        let title = frame.request.URL.host
+        let alertController = UIAlertController(title: title, message: message, preferredStyle: UIAlertControllerStyle.Alert)
+        alertController.addAction(UIAlertAction(title: OKString, style: UIAlertActionStyle.Default, handler: { _ in
+            completionHandler()
+        }))
+        presentViewController(alertController, animated: true, completion: nil)
+    }
+
+    func webView(webView: WKWebView, runJavaScriptConfirmPanelWithMessage message: String, initiatedByFrame frame: WKFrameInfo, completionHandler: (Bool) -> Void) {
+        // Show JavaScript confirm dialogs.
+        let title = frame.request.URL.host
+        let alertController = UIAlertController(title: title, message: message, preferredStyle: UIAlertControllerStyle.Alert)
+        alertController.addAction(UIAlertAction(title: OKString, style: UIAlertActionStyle.Default, handler: { _ in
+            completionHandler(true)
+        }))
+        alertController.addAction(UIAlertAction(title: CancelString, style: UIAlertActionStyle.Cancel, handler: { _ in
+            completionHandler(false)
+        }))
+        presentViewController(alertController, animated: true, completion: nil)
+    }
+
+    func webView(webView: WKWebView, runJavaScriptTextInputPanelWithPrompt prompt: String, defaultText: String?, initiatedByFrame frame: WKFrameInfo, completionHandler: (String!) -> Void) {
+        // Show JavaScript input dialogs.
+        let title = frame.request.URL.host
+        let alertController = UIAlertController(title: title, message: prompt, preferredStyle: UIAlertControllerStyle.Alert)
+        var input: UITextField!
+        alertController.addTextFieldWithConfigurationHandler({ (textField: UITextField!) in
+            textField.text = defaultText
+            input = textField
+        })
+        alertController.addAction(UIAlertAction(title: OKString, style: UIAlertActionStyle.Default, handler: { _ in
+            completionHandler(input.text)
+        }))
+        alertController.addAction(UIAlertAction(title: CancelString, style: UIAlertActionStyle.Cancel, handler: { _ in
+            completionHandler(nil)
+        }))
+        presentViewController(alertController, animated: true, completion: nil)
+    }
```

Comment:
Good question -- it shows the dialog, but the selected tab doesn't change (same behavior as Safari). To be consistent with desktop/FxA, we should probably switch to the alerting tab. I'll file a follow-up.